### PR TITLE
Backport AllocationService related patches

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

See commit messages for details

These are older patches missing from es-backports, motivated by recent flaky tests.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java [hunk 1]
```diff
@@ -200,7 +204,16 @@ public int getSize() {
     }
 
     /**
-     * Returns a {@link List} of assigned shards
+     * Returns a {@link List} of active shards
+     *
+     * @return a {@link List} of shards
+     */
+    public List<ShardRouting> getActiveShards() {
```
Comment (seut) [inline server/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java:211]: is this used somewhere? or what is the reason to add it?

**Comment by seut** [comment]
Could we had the original ES commit to this backport? https://github.com/crate/crate/pull/10769/commits/e8ccf07844b011922f20c68cbb044663cd018d87

**Review by seut — APPROVED** [review]
Lgtm otherwise

**Comment by mfussenegger** [comment]
> Could we had the original ES commit to this backport commit message? [e8ccf07](https://github.com/crate/crate/commit/e8ccf07844b011922f20c68cbb044663cd018d87)

Yep added it.

File: server/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java]: You are right it is not used, I accidentlly included with a cherry-pick

**Pull request merged**